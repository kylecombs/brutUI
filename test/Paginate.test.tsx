import * as React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Paginate } from "../src";

afterEach(cleanup);

describe("Pagination", () => {
  it("Renders mobile pagination correctly", () => {
    const { asFragment } = render(
      <Paginate page={0} setPage={() => {}} totalPages={10} isMobile />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders desktop pagination correctly", () => {
    const { asFragment } = render(
      <Paginate page={0} setPage={() => {}} totalPages={10} isMobile={false} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Can go to a different page on mobile", () => {
    let page = 0;

    const { rerender, getByTestId } = render(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile
      />,
    );

    // starts at 0, but cannot go back
    fireEvent.click(getByTestId("prevPage"));
    fireEvent.click(getByTestId("nextPage"));

    rerender(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile
      />,
    );

    expect(page).toBe(1);

    fireEvent.click(getByTestId("prevPage"));

    rerender(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile
      />,
    );

    expect(page).toBe(0);
  });

  it("Should not move past totalPages", () => {
    let page = 10;

    const { rerender, getByTestId } = render(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile
      />,
    );

    fireEvent.click(getByTestId("nextPage"));

    rerender(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile
      />,
    );

    expect(page).toBe(10);
  });

  it("Can go to a different page on desktop", () => {
    let page = 0;

    const { rerender, getByTestId } = render(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile={false}
      />,
    );

    fireEvent.click(getByTestId("nextPage"));

    rerender(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile={false}
      />,
    );

    expect(page).toBe(1);

    fireEvent.click(getByTestId("prevPage"));

    rerender(
      <Paginate
        page={page}
        setPage={(newPage: number) => {
          page = newPage;
        }}
        totalPages={10}
        isMobile={false}
      />,
    );

    expect(page).toBe(0);
  });
});
