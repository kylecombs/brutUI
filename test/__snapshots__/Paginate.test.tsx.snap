// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Renders desktop pagination correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full h-10 text-sm select-none"
  >
    <button
      class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none opacity-50"
      data-testid="prevPage"
      disabled=""
    >
      <svg
        class="mr-3"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="19"
          x2="5"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 19 5 12 12 5"
        />
      </svg>
      Previous
    </button>
    <div
      class="flex items-center justify-center flex-grow"
    >
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium bg-primary-50 dark:bg-opacity-0 text-primary-600 dark:text-white"
      >
        1
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
      >
        2
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
      >
        3
      </span>
      <span
        class="w-10 px-0.5 text-center dark:text-gray-500"
      >
        ...
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
      >
        9
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
      >
        10
      </span>
    </div>
    <button
      class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none cursor-pointer"
      data-testid="nextPage"
    >
      Next
      <svg
        class="ml-3"
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 5 19 12 12 19"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`Pagination Renders mobile pagination correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
  >
    <svg
      class="mr-3 text-gray-500 dark:text-white opacity-50"
      data-testid="prevPage"
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="19"
        x2="5"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 19 5 12 12 5"
      />
    </svg>
    <div
      class="flex justify-center flex-grow text-sm text-gray-700 select-none dark:text-white"
    >
      Page 1 out of 10
    </div>
    <svg
      class="ml-3 text-gray-500 dark:text-white cursor-pointer"
      data-testid="nextPage"
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </div>
</DocumentFragment>
`;
